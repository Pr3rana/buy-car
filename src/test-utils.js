import React from 'react';
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Switch, Link } from 'react-router-dom'; 
import { ListPageContext } from "./helpers/appContext";

export const carData = {
    cars: [{"stockNumber":10000,"manufacturerName":"Tesla","modelName":"Model S","color":"silver","mileage":{"number":151884,"unit":"km"},"fuelType":"Diesel","pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"}],
    totalPageCount:100,
    totalCarsCount:1000
}
export const isFetched = false;
const setTotalCarsCount = jest.fn();
const setTotalPageCount = jest.fn();
const setPagenumber = jest.fn();
const setSelectedFilter = jest.fn();
const setSavedCars = jest.fn();
const setAvailableFilters = jest.fn();
const setFilterParams = jest.fn();

const pageContextProvider = {totalCarsCount:1,
    setTotalCarsCount,
    totalPageCount:2,
    setTotalPageCount,
    pagenumber:1,
    setPagenumber,
    selectedFilter:[{}],
    setSelectedFilter,
    savedCars:[],
    setSavedCars,
    availableFilters:{test:["test value"]},
    setAvailableFilters,
    filterParams: [{test:"test filter value"}],
    setFilterParams
}

const AllTheProviders = ({ children }) => {
    return (
     <Router>
          <ListPageContext.Provider value={pageContextProvider}>{children}</ListPageContext.Provider>
     </Router>
    )
}

const customRender = (ui, options) => {
    render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };