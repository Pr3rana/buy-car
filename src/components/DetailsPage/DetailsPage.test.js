import { render, screen, cleanup } from '../../test-utils.js';
import Details from './DetailsPage';

afterEach(cleanup)
describe("Render details",()=>{
    test("initial render",()=>{
        render(<Details/>)
        expect(screen.getByTestId('details-page')).toMatchSnapshot(
            `<div class="details-container"><article class="content-wrapper"><div class="banner-img">
            <img alt="banner" src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"></div>
            <div class="detail-wrapper"><summary class="details"><p class="header">Model S</p>
            <p class="product-key-details"><span>Stock # 10000 - </span><span>151884km - </span>
            <span>Diesel - </span><span>silver</span></p><p>This car is currently available and can be delivered as soon as tomorrow morning. 
            Please be aware that the delivery times shown in this page are not definitive and may change due to a bad weather conditions.</p>
            </summary><div class="details-favourite"><p>If you like the car, click the button and save it in your collection of favourite items.</p>
            <button data-testid="default-btn" class="default-btn ">Save</button></div></div></article></div>
            `
        )
        expect(screen.queryByTestId('details-loader')).toBeNull()
    })
})
