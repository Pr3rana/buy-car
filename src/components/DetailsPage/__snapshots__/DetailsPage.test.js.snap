// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render details initial render: <div class="details-container"><article class="content-wrapper"><div class="banner-img">
            <img alt="banner" src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"></div>
            <div class="detail-wrapper"><summary class="details"><p class="header">Model S</p>
            <p class="product-key-details"><span>Stock # 10000 - </span><span>151884km - </span>
            <span>Diesel - </span><span>silver</span></p><p>This car is currently available and can be delivered as soon as tomorrow morning. 
            Please be aware that the delivery times shown in this page are not definitive and may change due to a bad weather conditions.</p>
            </summary><div class="details-favourite"><p>If you like the car, click the button and save it in your collection of favourite items.</p>
            <button data-testid="default-btn" class="default-btn ">Save</button></div></div></article></div>
             1`] = `
<div
  class="details-container"
  data-testid="details-page"
>
  <div
    class="details-loader"
  >
    <img
      alt="Loader"
      src="https://i.gifer.com/ZZ5H.gif"
    />
  </div>
</div>
`;
